src/storage/mod.rs

```diff
@@ -957,25 +957,30 @@ impl<E: Engine, L: LockMgr> Storage<E, L> {
                                 ctx.get_isolation_level(),
                                 !ctx.get_not_fill_cache(),
                             );
-                            let kv_pairs: Vec<_> = snap_store
+                            let result = snap_store
                                 .batch_get(&keys, &mut statistics)
-                                .into_iter()
-                                .zip(keys)
-                                .filter(|&(ref v, ref _k)| {
-                                    !(v.is_ok() && v.as_ref().unwrap().is_none())
-                                })
-                                .map(|(v, k)| match v {
-                                    Ok(Some(x)) => Ok((k.into_raw().unwrap(), x)),
-                                    Err(e) => Err(Error::from(e)),
-                                    _ => unreachable!(),
-                                })
-                                .collect();
+                                .map_err(Error::from)
+                                .map(|v| {
+                                    let kv_pairs: Vec<_> = v
+                                        .into_iter()
+                                        .zip(keys)
+                                        .filter(|&(ref v, ref _k)| {
+                                            !(v.is_ok() && v.as_ref().unwrap().is_none())
+                                        })
+                                        .map(|(v, k)| match v {
+                                            Ok(Some(x)) => Ok((k.into_raw().unwrap(), x)),
```

Comment:
Could we just match `Ok(x)` since if we meet `Ok(None)`, it will panic.

Reply:
So you need to either write `Ok(Some(x)) => ...x` or `Ok(x) => ...x.unwrap()`? I think both are the same.